### PR TITLE
Ch mccl cleanup

### PR DESCRIPTION
**Pull request by hayakawa16**

### CODE REVIEW

File: src/Vts/MonteCarlo/AbsorptionWeightingMethods.cs [hunk 1]
```diff
@@ -136,10 +130,8 @@ private static double VolumeAbsorptionWeightingDiscrete(PhotonDataPoint previous
         private static double VolumeAbsorbAnalog(PhotonStateType photonStateType)
         {
             var weight = 0.0;
-            if (photonStateType.HasFlag(PhotonStateType.Absorbed))
-            {
-                weight = 1.0;
-            }
+            if (!photonStateType.HasFlag(PhotonStateType.Absorbed)) return weight;
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/AbsorptionWeightingMethods.cs:133]: The variable for weight seems redundant here, we are either returning 1.0 or 0.0, could we make this a return statement?

File: src/Vts/MonteCarlo/Controllers/HistoryDetectorController.cs [hunk 2]
```diff
@@ -19,7 +18,6 @@ public class HistoryDetectorController : IDetectorController
         public HistoryDetectorController(IEnumerable<IHistoryDetector> detectors, ITissue tissue)
         {
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Controllers/HistoryDetectorController.cs:19]: It appears that tissue doesn't do anything, I know we can't remove it but maybe we should keep a note of this so we can figure out what to do going forward.

File: src/Vts/MonteCarlo/DataStructures/SimulationInput.cs [hunk 3]
```diff
@@ -39,10 +38,7 @@ public class SimulationInput
             TissueInput = tissueInput;
             DetectorInputs = detectorInputs;       
             // check if detectorInputs list is null and if so make empty
-            if (DetectorInputs == null)
-            {
-                DetectorInputs = new List<IDetectorInput>() {};
-            }
+            if (DetectorInputs == null)  DetectorInputs = new List<IDetectorInput> {};
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/DataStructures/SimulationInput.cs:41]: This is not a return so should it be on one line? Was this a cleanup suggestion?

File: src/Vts/MonteCarlo/DataStructuresValidation/TissueInputs/BoundingCylinderTissueInputValidation.cs [hunk 4]
```diff
@@ -47,23 +43,24 @@ public static ValidationResult ValidateInput(ITissueInput input)
...
                     false,
                     "BoundingCylinderTissueInput: tissue layer is assumed to be at least a single layer with air layer above and below",
                     "BoundingCylinderTissueInput: redefine tissue definition to contain at least a single layer of tissue");
             }
 
-            if (!tempResult.IsValid) { return tempResult; }
+            if (!tempResult.IsValid) return tempResult;
 
-            var layersHeight = tissueLayers.Sum(layer => layer.ZRange.Delta);
+            var layersHeight = layerTissueRegions.Sum(layer => layer.ZRange.Delta);
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/DataStructuresValidation/TissueInputs/BoundingCylinderTissueInputValidation.cs:64]: Did the functionality of this code change?

File: src/Vts/MonteCarlo/DataStructuresValidation/TissueInputs/SingleEllipsoidTissueInputValidation.cs [hunk 5]
```diff
@@ -47,34 +43,35 @@ private static ValidationResult ValidateGeometry(IList<LayerTissueRegion> layers
...
                     "SingleEllipsoidTissueInput: redefine tissue definition to contain at least a single layer of tissue");
             }
 
-            if (!tempResult.IsValid) { return tempResult; }
+            if (!tempResult.IsValid) return tempResult;
 
             // check that ellipsoid contained within a tissue layer
-            bool correctlyContainedInLayer = tissueLayers.Any(
+            var correctlyContainedInLayer = layerTissueRegions.Any(
                 layer =>
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/DataStructuresValidation/TissueInputs/SingleEllipsoidTissueInputValidation.cs:75]: Same question as above - did this code functionality change?

File: src/Vts/MonteCarlo/Detectors/Detector.cs [hunk 6]
```diff
@@ -8,7 +8,7 @@ public abstract class DetectorInput
         /// <summary>
         /// default constructor for detector input
         /// </summary>
-        public DetectorInput()
+        protected DetectorInput()
         {
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Detectors/Detector.cs:12]: What was this cleanup suggestion?

File: src/Vts/MonteCarlo/DataStructures/SimulationInput.cs (hunk 3 above)
Comment (lmalenfant) [reply src/Vts/MonteCarlo/DataStructures/SimulationInput.cs:41]: I saw this later in the cleanup and I think on one line is ok, I just had a question about whether we should have removed the ( ) or the { }

File: src/Vts/MonteCarlo/AbsorptionWeightingMethods.cs (hunk 1 above)
Comment (hayakawa16) [reply src/Vts/MonteCarlo/AbsorptionWeightingMethods.cs:133]: Removed weight and modified into ? : statement, and added header for description.

File: src/Vts/MonteCarlo/Controllers/HistoryDetectorController.cs (hunk 2 above)
Comment (hayakawa16) [reply src/Vts/MonteCarlo/Controllers/HistoryDetectorController.cs:19]: Added parameter description to state parameter "tissue" not used.

File: src/Vts/MonteCarlo/DataStructures/SimulationInput.cs (hunk 3 above)
Comment (hayakawa16) [reply src/Vts/MonteCarlo/DataStructures/SimulationInput.cs:41]: Replaced "{}" with "()"

File: src/Vts/MonteCarlo/DataStructuresValidation/TissueInputs/BoundingCylinderTissueInputValidation.cs (hunk 4 above)
Comment (hayakawa16) [reply src/Vts/MonteCarlo/DataStructuresValidation/TissueInputs/BoundingCylinderTissueInputValidation.cs:64]: This one was okay as is.

File: src/Vts/MonteCarlo/DataStructuresValidation/TissueInputs/SingleEllipsoidTissueInputValidation.cs (hunk 5 above)
Comment (hayakawa16) [reply src/Vts/MonteCarlo/DataStructuresValidation/TissueInputs/SingleEllipsoidTissueInputValidation.cs:75]: This one is okay as is.

File: src/Vts/MonteCarlo/Detectors/Detector.cs (hunk 6 above)
Comment (hayakawa16) [reply src/Vts/MonteCarlo/Detectors/Detector.cs:12]: Okay since abstract class.

File: src/Vts/MonteCarlo/DataStructures/SimulationOptions.cs [hunk 7]
```diff
@@ -36,10 +36,7 @@ public class SimulationOptions
             PhaseFunctionType = phaseFunctionType;
             Databases = databases;
             // check if databases list is null and if so make empty
-            if (Databases == null)
-            {
-                Databases = new List<DatabaseType>() { };
-            }
+            if (Databases == null) Databases = new List<DatabaseType> { };
             Seed = seed;
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/DataStructures/SimulationOptions.cs:40]: I missed this one when we went through, same as before, add ( ) and remove { }

Comment (hayakawa16) [reply src/Vts/MonteCarlo/DataStructures/SimulationOptions.cs:40]: Removed "{}" and added "()".

**Review by lmalenfant — APPROVED** [review]

File: src/Vts/MonteCarlo/DataStructures/SimulationOptions.cs (hunk 7 above)
Comment (hayakawa16) [reply src/Vts/MonteCarlo/DataStructures/SimulationOptions.cs:40]: Made the change and pushed.

**Pull request merged**